Initial commit of the Data attribute which exposes various data about the server and its neighborhood (#56)
many-protocol/specification: attributes/network/5_data.adoc

= Data Attribute (#6)
:cddl: ./cddl/

Any server that implements this attribute also provides a list of data points (either scalar or vectors) that can be queried by clients.
The data attribute should be used to provide statistics, gauges and general queryable information about the server, its neighborhood and its network.

Notably, those data aren't meant to be gathered and reduced in any way.
They're information about the network, and not metrics for monitoring.

All data are specified in this specification, and are considered informative on the node/neighborhood, rather than monitoring.
Servers MAY NOT create new data types without being a specification.

== Examples

> A data namespace `[0, ...]` contains the number of identities or addresses the server has seen in a certain context.
> The server should disclose the `[5, [0]]` attribute with its argument.
> For the purpose of this example, this data namespace has one sub namespace: `[0, [2, ...]]`, which is ledger specific.
> The server is allowed to skip the ledger specific types if it does not implement the ledger attribute.

Reviewer: ```suggestion
A data index `[0, ...]` contains the number of identities or addresses the server has seen in a specific context.
The server should disclose the `[5, [0]]` attribute with its argument.
This data index has one sub-index for this example: `[0, [2, ...]]`, which is ledger specific.
The server can skip the ledger-specific types if it does not implement the ledger attribute.
```

> All of these names and their types are specified in their own specification file in the `./5_data/` folder.

Reviewer: ```suggestion
These indices and types are specified in their own specification file in the `./5_data/` folder.
```
